client/state/data-layer/wpcom/me/settings/index.js

```diff
@@ -29,7 +29,11 @@ function fromApi( apiResponse ) {
 
 	// Some keys in the `decodedValues` can be undefined, and _.merge will ignore them,
 	// while Object.assign or object spread operator wouldn't.
-	return merge( {}, apiResponse, decodedValues );
+	return merge(
+		{},
+		omitBy( apiResponse, isUndefined ),
+		omitBy( decodedValues, isUndefined )
+	);
```

Comment:
I used `merge` here to ignore the `undefined` values that the `decodedValues` code produces in place of missing properties:
```js
const decodedValues = {
  display_name: apiResponse.display_name && decodeEntities(apiResponse.display_name)
}
```
If we remove these properties with `omitBy`, we can avoid `merge` altogether and use just object spread operator (that compiles to `Object.assign`) which is more obvious to figure out:
```js
const decodedValues = omitBy( {
  display_name: ...
}, isUndefined );

return {
  ...apiResponse, // keep all properties intact, no need to omit anything
  ...decodedValues, // has only the defined ones already
};
```